frontend/src/components/messages/hooks/useMessages.ts

```diff
@@ -0,0 +1,52 @@
+import { SetStateAction, useEffect, useState } from "react";
+
+const UseMessages = () => {
+    const [messages, setMessages] = useState<
+        { id: string; message: string; author: string }[]
+    >([]);
+
+    useEffect(() => {
+        let ignore = true;
```

Comment:
ignore is true on every useEffect loop, u set message on every useEffect :)

Reply:
It's fix for race condition, I'd leave it for now